### PR TITLE
Update Typespec for Language Conversations API

### PR DESCRIPTION
**Pull request by quentinRobinson**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [X] a public preview
- [X] GA release

### Change Scope
Moving to typsespec. fixed some errors in tsp and updated the LRO operations to be able to generate the SDK

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/cognitiveservices/Language.Conversations/common.tsp [hunk 1]
```diff
@@ -535,31 +543,45 @@ model CustomResult {
 @doc("Optional parameter to use a Custom Character to be used for redaction in PII responses. Default character will be * as before. We allow specific ascii characters for redaction.")
 enum RedactionCharacter{
     #suppress "@azure-tools/typespec-azure-core/documentation-required"
-    `!`,
+    explationPoint: "!",
```
Comment (heaths) [inline specification/cognitiveservices/Language.Conversations/common.tsp:546]: You mean `exclamationPoint`?

File: specification/cognitiveservices/Language.Conversations/common.tsp [hunk 2]
```diff
@@ -535,31 +543,45 @@ model CustomResult {
...
-    `#`,
+    numberSign: "#",
+
     #suppress "@azure-tools/typespec-azure-core/documentation-required"
-    `$`,
+    dollar: "$",
+
     #suppress "@azure-tools/typespec-azure-core/documentation-required"
-    `%`,
+    perCent: "%",
```
Comment (heaths) [inline specification/cognitiveservices/Language.Conversations/common.tsp:555]: It's a single word: `percent`.

File: specification/cognitiveservices/Language.Conversations/main.tsp [hunk 3]
```diff
@@ -1,15 +1,15 @@
 import "@typespec/rest";
 import "@typespec/http";
 import "@typespec/versioning";
-import "@typespec/openapi";
+// import "@typespec/openapi";
```
Comment (heaths) [inline specification/cognitiveservices/Language.Conversations/main.tsp:4]: Given it's in source control, best to remove unless this truly is temporary.

Comment (heaths) [reply specification/cognitiveservices/Language.Conversations/main.tsp:4]: Same for other instances below.

File: specification/cognitiveservices/Language.Conversations/main.tsp [hunk 4]
```diff
@@ -586,29 +586,28 @@ model AnalyzeConversationJobsInput {
...
-  tasks: Array<AnalyzeConversationLROTask>;
+  tasks: Array<AnalyzeConversationLroTask>;
 }
 
 #suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use PascalCase"
 @doc("Base class for a long-running conversation input task.")
 @discriminator("kind")
 @added(Versions.v2023_04_01)
-model AnalyzeConversationLROTask extends TaskIdentifier {
+model AnalyzeConversationLroTask extends TaskIdentifier {
```
Comment (heaths) [inline specification/cognitiveservices/Language.Conversations/main.tsp:602]: The uppercase "LRO" should be correct. Language generators should do the right thing for generated code. Were you not seeing that, though?

File: specification/cognitiveservices/Language.Conversations/main.tsp [hunk 5]
```diff
@@ -647,9 +647,9 @@ enum AnalyzeConversationResultsKind {
 @doc("Contains the status of the submitted job for analyzing a conversation, along with related statistics.")
 @added(Versions.v2023_04_01)
 @resource("language/analyze-conversations/jobs")
-model AnalyzeConversationJobState {
+model AnalyzeConversationJobState{
```
Comment (heaths) [inline specification/cognitiveservices/Language.Conversations/main.tsp:650]: Nit: should retain the space for consistency.

File: specification/cognitiveservices/Language.Conversations/main.tsp [hunk 6]
```diff
@@ -844,12 +849,12 @@ model ConversationItemLevelTiming extends AudioTiming {}
 
 @added(Versions.v2023_04_15_preview)
 @doc("Task definition for conversational sentiment.")
-model AnalyzeConversationalSentimentTask extends AnalyzeConversationLROTask {
+model AnalyzeConversationalSentimentTask extends AnalyzeConversationLroTask {
```
Comment (heaths) [inline specification/cognitiveservices/Language.Conversations/main.tsp:852]: Same here for "LRO".

File: specification/cognitiveservices/Language.Conversations/main.tsp [hunk 7]
```diff
@@ -878,8 +883,9 @@ model ConversationSentimentResults extends PreBuiltResult{
 
 @added(Versions.v2023_04_15_preview)
 @doc("The result from sentiment analysis operation for each conversation.")
-model ConversationSentimentResultsConversation extends ConversationSentimentResult{
-  ...ConversationResultBase
+model ConversationSentimentResultsConversation{
```
Comment (heaths) [inline specification/cognitiveservices/Language.Conversations/main.tsp:886]: Nit:
```suggestion
model ConversationSentimentResultsConversation {
```

Comment (heaths) [reply specification/cognitiveservices/Language.Conversations/main.tsp:886]: I won't repeat, but some other instances below. Generally, we like to keep things "open" - makes it more reviewable.

**Review by heaths — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/Language.Conversations/main.tsp (hunk 4 above)
Comment (quentinRobinson) [reply specification/cognitiveservices/Language.Conversations/main.tsp:602]: it was throwing a PascalCase warning. Ill suppress the warning.

File: specification/cognitiveservices/Language.Conversations/main.tsp [hunk 8]
```diff
@@ -649,7 +646,7 @@ enum AnalyzeConversationResultsKind {
 @resource("language/analyze-conversations/jobs")
 model AnalyzeConversationJobState {
```
[the remark this answers is not in the document]
Comment (heaths) [reply specification/cognitiveservices/Language.Conversations/main.tsp:647]: Interesting. @johanste, @bterlson is that a bug?

**Pull request closed without merging**